synapse_invite_checker/invite_checker.py

```diff
@@ -667,17 +675,32 @@ async def on_create_room(
                 errors.Codes.FORBIDDEN,
             )
 
+        # preset can be any of "private_chat", "trusted_private_chat" or "public_chat"
+        # Do not allow "public_chat". Default is based on setting of visibility
+        room_preset: str = request_content.get("preset")
+        # visibility can be either "public" or "private". If not included, it defaults to "private"
+        room_visibility: str = request_content.get("visibility", "private")
+        # Determine based on above that the room is probably public
+        is_public = (
```

Comment:
what if I put join rules in the initial state?

Reply:
That is not what I am asking. I was asking what happens if you put a join rule into the initial_state list of the /createRoom call.

Reply:
Ah, the code was folded and I didn't notice it, seems fine! :)